components/MultiDeleteModal.vue

```diff
@@ -0,0 +1,260 @@
+      columns                : [{ key: 'name' }, { key: 'organization' }, { key: 'emailOrCountry' }],
+      page                   : 1,
+      perPage                : 5,
+      recordCount            : this.selectionList.length,
+      deleteRecordCount      : 0,
+      confirmationRecordCount: '',
+      errorMsgDismissTtl     : 0,
+      stopDeletion           : false,
+    };
+  },
+  computed: {
+    title()         { return `Deletion of ${this.recordCount} records`; },
+    deleteSuccess() { return this.deleteRecordCount === this.recordCount; },
+    progressValue() { return  Math.round((this.deleteRecordCount / this.recordCount) * 100); },
+    items()         {
+      return this.selectionList.map((s) => ({
+        name          : getName(s).trim(),
+        organization  : getOrganization.call(this, s).trim(),
+        emailOrCountry: getEmailOrCountry.call(this, s).trim(),
+      }), this);
+    },
+    ...mapGetters({
+      getCountryNameByCode   : 'countries/getNameByCode',
+      getOrganizationTypeById: 'organizations/getTypeById',
+    }),
+  },
+  methods: {
+    close,
+    onDelete,
+    deleteContact,
+    deleteOrganization,
+    ...mapActions({
+      clearContactsSelection     : 'contacts/clearSelection',
+      clearOrganizationsSelection: 'organizations/clearSelection',
+    }),
+  },
+};
+
+function close() {
+  this.step = 'preview';
+  this.show = false;
+  this.$emit('update:show', this.show);
+}
+
+function getName(entry) {
+  if (entry.contactId) return `${entry.firstName} ${entry.lastName}`;
+
+  return `${entry.acronym || entry.name}`;
+}
+
+function getOrganization(entry) {
+  if (entry.contactId) return `${entry.organizationAcronym || entry.organizationName}`;
+
+  const  organizationType = this.getOrganizationTypeById(entry.organizationTypeId);
+
+  return `${organizationType.acronym || organizationType.title}`;
+}
+
+function getEmailOrCountry(entry) {
+  if (entry.contactId) return `${entry.emails.length ? entry.emails[0] : ''}`;
+
+  return `${this.getCountryNameByCode(entry.government || entry.country)}`;
+}
+
+async function onDelete() {
+  try {
+    if (this.recordCount !== this.confirmationRecordCount) {
+      this.errorMsgDismissTtl = 3;
+      return;
+    }
+
+    this.step              = 'progress';
+    this.deleteRecordCount = 0;
+
+    if ('contactId' in this.selectionList[0]) {
+      await this.deleteContact();
+    } else {
+      await this.deleteOrganization();
+    }
+  } catch (error) {
+    this.errorMsgDismissTtl = 3;
+  }
+}
+
+async function deleteContact() {
+  for (let index = 0; index < this.selectionList.length; index++) {
+    if (this.stopDeletion) break;
+
+    // eslint-disable-next-line no-await-in-loop
+    await this.$kronosApi.deleteContact(this.selectionList[index].contactId);
+    this.deleteRecordCount++;
+  }
+
+  if (this.recordCount === this.deleteRecordCount) {
+    this.$root.$emit('bv::refresh::table', 'KContactsList');
+    this.clearContactsSelection();
+  }
```

Comment:
You cannot assume that you are on a page where where contact list will be there. and if another control need to be refreshed you will have to handled it here?

better to emit a generic event that other component can listen to
eg: 
```suggestion
  if (this.recordCount === this.deleteRecordCount) {
    this.$root.$emit('record-deleted');
    this.clearContactsSelection();
  }
```

then in the contactList, if the event is triggered then you refresh the result

Reply:
Done!